pulpcore/cli/core/export.py

```diff
@@ -0,0 +1,131 @@
+from pulpcore.cli.common import (
+    DEFAULT_LIMIT,
+    PulpContext,
+    PulpEntityContext,
+    RepositoryVersionDefinition,
+)
+
+from typing import Any, Dict, List
+import click
+
+
+class ExportContext(PulpEntityContext):
+    ENTITY: str = "Export"
+
+
+class PulpExportContext(ExportContext):
+    ENTITY: str = "PulpExport"
+    HREF: str = "core_pulp_pulp_export_href"
+    READ_ID: str = "exporters_core_pulp_read"
+    DELETE_ID: str = "exporters_core_pulp_exports_delete"
+    LIST_ID: str = "exporters_core_pulp_exports_list"
+    EXPORTER_LIST_ID: str = "exporters_core_pulp_list"
+    CREATE_ID: str = "exporters_core_pulp_exports_create"
+
+    def find_exporter(self, name: str) -> Any:
+        search_result = self.pulp_ctx.call(
+            self.EXPORTER_LIST_ID, parameters={"name": name, "limit": 1}
+        )
+        if search_result["count"] != 1:
+            raise click.ClickException(f"PulpExporter '{name}' not found.")
+        exporter = search_result["results"][0]
+        return exporter
+
+
+@click.group()
+@click.pass_context
+def export(ctx: click.Context) -> None:
+    pulp_ctx: PulpContext = ctx.find_object(PulpContext)
+    ctx.obj = ExportContext(pulp_ctx)
+
+
+@export.group()
+@click.pass_context
+def pulp(ctx: click.Context) -> None:
```

Comment:
Is this level of subcommand needed? As i read it, it will make all the commands look like `pulp export pulp <the real thing>`. Are we ever going to have something else than "pulp" exports?

Reply:
Yes - FilesystemExporter already exists, and i think there has been at least one other proposed that I can't recall right now.

Reply:
So it will be like
```
pulp export filesystem <...>
pulp export pulp <...>
pulp export rsync <...>
...
```

Reply:
I was thinking it would be like:

```
pulp exporter pulp export list --exporter MyExporter
pulp exporter pulp export create --exporter MyExporter
...
```

Reply:
I'm fine with `pulp export <type>` though since it's less typing.

Reply:
--exporter is a fine idea, +1